pkg/artifactory/resource_xray_watch.go

```diff
@@ -274,6 +275,11 @@ func resourceXrayWatchRead(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).Xray
 
 	watch, resp, err := c.V2.Watches.GetWatch(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.